Softeq.XToolkit.WhiteLabel/Model/TabItem.cs

```diff
@@ -1,37 +1,41 @@
 // Developed by Softeq Development Corporation
 // http://www.softeq.com
 
+using Softeq.XToolkit.WhiteLabel.Bootstrapper.Abstract;
 using Softeq.XToolkit.WhiteLabel.Mvvm;
 using Softeq.XToolkit.WhiteLabel.Navigation;
 using Softeq.XToolkit.WhiteLabel.ViewModels.Tab;
 
 namespace Softeq.XToolkit.WhiteLabel.Model
 {
-    public abstract class TabItem
+    public abstract class TabItem<TKey>
     {
-        protected TabItem(string title, string imageName)
+        protected TabItem(string title, TKey key)
         {
             Title = title;
-            ImageKey = imageName;
+            Key = key;
         }
 
         public string Title { get; }
 
-        public string ImageKey { get; }
+        public TKey Key { get; }
 
-        public abstract TabViewModel CreateViewModel();
+        public abstract TabViewModel<TKey> CreateViewModel();
     }
 
-    public class TabItem<TFirstViewModel> : TabItem where TFirstViewModel : ViewModelBase
+    public class TabItem<TFirstViewModel, TKey> : TabItem<TKey> where TFirstViewModel : ViewModelBase
     {
-        public TabItem(string title, string imageName) : base(title, imageName)
+        public TabItem(string title, TKey key, IContainer container) : base(title, key)
         {
+            Container = container;
         }
 
-        public override TabViewModel CreateViewModel()
+        protected IContainer Container { get; }
+
+        public override TabViewModel<TKey> CreateViewModel()
         {
-            var frameNavigationService = Dependencies.Container.Resolve<IFrameNavigationService>();
-            var tabViewModel = new TabViewModel<TFirstViewModel>(frameNavigationService);
+            var frameNavigationService = Container.Resolve<IFrameNavigationService>();
```

Comment:
As idea: create a separate abstraction for getting `IFrameNavigationService`.

We should avoid using `IContainer` explicitly in high-level classes.

In this case, I prefer to keep using `Dependencies`.